packages/discovery-react-components/src/components/SearchResults/components/Result/Result.tsx

```diff
@@ -102,26 +101,32 @@ export const Result: React.FunctionComponent<ResultProps> = ({
     fetchDocumentsResponseStore: { isLoading }
   } = useContext(SearchContext);
 
-  const firstPassage: DiscoveryV2.QueryResultPassage | undefined = get(
+  const passages: DiscoveryV2.QueryResultPassage[] | undefined = get(
     result,
-    'document_passages[0]'
+    'document_passages'
+  )?.filter(passage => !!get(passage, 'passage_text'));
+  const passageTexts: (string | undefined)[] | undefined = passages?.map(passage =>
```

Comment:
So this can be an array of `undefined`? That doesn't seem right.

Reply:
This is because `get` could technically return `undefined`. (Even with optional chaining, TS will think that this could return `undefined`.) However, we know that it can't since we already filtered out that possibility on line 107.

Reply:
I tried to build out an interface that extended `DiscoveryV2.QueryResultPassage`, but required `passage_text`, but hit a snag (forgetting exactly what). I'll try that again and see how it goes this time...

Reply:
`!` operator is useful here as well. :+1: